### PR TITLE
[Cross-entropy-loss] return mean token accuracy metric with CE loss

### PR DESCRIPTION
**Pull request by kashif**
## Summary

Returns the mean token accuracy metric when minimizing the cross-entropy loss without materializing the logits

https://x.com/jeremyphoward/status/1703246293802586155

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/functional.py [hunk 1]
```diff
@@ -47,10 +48,20 @@ def liger_cross_entropy(
...
         return loss
-    return loss, z_loss
+
+    result = [loss]
+    if return_z_loss:
+        result.append(z_loss)
+    if return_token_accuracy:
+        result.append(token_accuracy)
+
+    return tuple(result) if len(result) > 1 else result[0]
```
Comment (albertvillanova) [inline src/liger_kernel/transformers/functional.py]: I see you tried to align with the code already present before your PR, but I find a potential ambiguity here: when a 2-tuple is returned, its 2nd element might be:
- `z_loss`: `return_z_loss=True, return_token_accuracy=False`
- `token_accuracy`: `return_z_loss=False, return_token_accuracy=True`

File: src/liger_kernel/transformers/functional.py [hunk 2]
```diff
@@ -82,10 +94,20 @@ def liger_fused_linear_cross_entropy(
...
```
Comment (albertvillanova) [inline src/liger_kernel/transformers/functional.py]: The same here.

File: src/liger_kernel/transformers/model/falcon_h1.py [hunk 3]
```diff
@@ -85,24 +86,32 @@ def lce_forward(
...
             labels=labels,
             shift_labels=shift_labels,
             hidden_size=self.config.hidden_size,
             **kwargs,
         )
+        # Unpack loss and token_accuracy if returned as tuple
+        if isinstance(result, tuple):
+            loss, token_accuracy = result
+        else:
+            loss = result
```
Comment (albertvillanova) [inline src/liger_kernel/transformers/model/falcon_h1.py]: The code before your PR assumed that `return_z_loss` was not passed within the `kwargs`: they assumed only the loss scalar was returned.
- Do you know if this assumption is justified?
- If not, then the returned type could be a 2 or 3-tuple (besides the scalar loss), with z_loss and/or token_accuracy

**Comment by kashif** [comment]
@vaibhavjindal would you be able to kindly review?

**Comment by kashif** [comment]
@shimizust this will be a breaking change i believe BTW

**Comment by vaibhavjindal** [comment]
@kashif could you please elaborate on how it will be a breaking change? Will it break the intergration with transformers or trl?

**Comment by kashif** [comment]
yes if someone is using the raw functions in their lib. then now that functions returns one more thing... but on the HF side this PR takes care of this

**Comment by kashif** [comment]
@vaibhavjindal  see here https://github.com/linkedin/Liger-Kernel/pull/910/files#diff-7654a885e261ec4986c229d953681635ab4c8761c89342d2dde604f169783b35L350

**Comment by vaibhavjindal** [comment]
@kashif got it. So if i understand correctly, it will make sure that liger remains compatible with newer versions from HF. However, just want to confirm it will break liger support with older transformers/trl versions?

**Comment by kashif** [comment]
no i believe my changes here will work with older version of HF.. i just meant non-HF frameworks

**Comment by kashif** [comment]
TRL relies on HF integration for the CE loss so in TRL I will just pin to the liger version that has these changes

**Comment by kashif** [comment]
@vaibhavjindal  let me fix up the new qwen3-vl model to update its API

**Comment by kashif** [comment]
@vaibhavjindal all good from my side

**Comment by vaibhavjindal** [comment]
> @vaibhavjindal all good from my side

Thanks a lot! I will do some final checks on correctness and benchmarks and will try to get it merged soon.

**Comment by kashif** [comment]
thank you so much.. also see here: https://github.com/huggingface/trl/pull/4302#issuecomment-3487634176

**Comment by kashif** [comment]
thanks @vaibhavjindal  for the typo fix and making it more robust!
